api/api-doc/storage_service.json

```diff
@@ -2324,6 +2359,103 @@
                "description":"The endpoint details"
             }
          }
+      },
+      "map_entry":{
+        "id":"map_entry",
+        "properties":{
+            "key":{
+                "type":"string"
+            },
+            "value":{
+                "type":"string"
+            }
+        }
+      },     
+      "named_maps":{
+        "id":"named_maps",
+        "properties":{
+            "group":{
+                "type":"string"
+            },
+            "attributes":{
+                "type":"array",
+                "items":{
+                    "$ref":"map_entry"
```

Comment:
if you use type instead of $ref you'll not need to change the json2code

Reply:
Missed that.

Reply:
are you sure it's not legal in 1.2?
anyhow, we do use it in other places, so you can use it as well.

Reply:
Ok, why not. Still think it is an easy fix to the generator code, since it already handles $ref for actual code gen (only dependency checking broke). But I've changed it to "type".